src/routers/alpha-router/alpha-router.ts

```diff
@@ -220,9 +220,18 @@ export type ProtocolPoolSelection = {
   /**
    * Given the topNTokenInOut pools, gets the top N pools that involve the other token.
    * E.g. for a WETH -> USDC swap, if topNTokenInOut found WETH -> DAI and WETH -> USDT,
-   * a value of 2 would find the top 2 pools that involve DAI or USDT.
+   * a value of 2 would find the top 2 pools that involve DAI and top 2 pools that involve USDT.
    */
   topNSecondHop: number;
+  /**
+   * Given the topNTokenInOut pools and a token address,
+   * gets the top N pools that involve the other token.
+   * If token address is not on the list, we default to topNSecondHop.
+   * E.g. for a WETH -> USDC swap, if topNTokenInOut found WETH -> DAI and WETH -> USDT,
+   * and there's a mapping USDT => 4, but no mapping for DAI
+   * it would find the top 4 pools that involve USDT, and find the topNSecondHop pools that involve DAI
+   */
+  topNSecondHopForTokenAddress?: Map<string, number>;
```

Comment:
Can we add something to make the key not case sensitive?

Reply:
good point, do we have a function somewhere that normalizes the key?